core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -46,7 +46,36 @@ object Automation {
         return rank
     }
 
-
+    // More complicated logic to properly weigh Food vs other Stats (esp Production)
+    private fun getFoodModWeight(city: City, surplusFood: Float): Float {
+        var foodModWeight = 1f
+
+        // Zero out Growth if close to Unhappiness limit
+        if (city.civ.getHappiness() < -8) {
+            foodModWeight = 0f
```

Comment:
Change these to "returns" to have minimal nesting :)
I see there's no actual modifications ( +, -) only setting (=) so should be trivial
Replace the final "return" with "return 1f" for default case, and then each specific case gets a "return" as well

Reply:
Done